pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -41,6 +41,8 @@ const (
 )
 
 type kubenetNetworkPlugin struct {
+	network.NoopNetworkPlugin
```

Comment:
It looks to me that kubenet already support shaping
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/network/kubenet/kubenet_linux.go#L211

Just include NET_PLUGIN_CAPABILITY_SHAPING for kubenet would work